site/develop/design/lazy_filter.md

```diff
@@ -0,0 +1,352 @@
+* [Lazy Filter](#LazyFilter)
```

Comment:
Thank you for adding this doc. Let's add this separately in order to focus the implementation.
Usually, the doc becomes obsolete during reviews. So, we had better focus on the document fully after code commit.

Reply:
Sure will pull this out into a separate PR